### PR TITLE
Feature/te5 t625 android web services

### PR DESCRIPTION
**Pull request by buddhdev-jay**

### CODE REVIEW

File: app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import kotlinx.android.synthetic.main.activity_login.editxt_password_login
+import kotlinx.android.synthetic.main.activity_login.progressbar_login
+import org.json.JSONObject
+
+
+class LoginActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+        val prefs = getSharedPreferences("sharedPrefLogin", Context.MODE_PRIVATE)
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt]: static key "sharedPrefLogin"

File: app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                            val result = sendPostRequest(
+                                editxt_email_login.text.toString(),
+                                editxt_password_login.text.toString()
+                            )
+                            if (result) {
+                                intent = Intent(applicationContext, MainActivity::class.java)
+                                startActivity(intent)
+                                isLogged = true
+                                val editor = prefs.edit()
+                                editor.putBoolean("lockedState", isLogged)
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt]: static key

File: app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            writer.write(cred.toString())
+            writer.flush()
+            println("URL : $url")
+            println("cred: $cred")
+            println("Response Code: $responseCode")
+            if (responseCode == 200) {
+                runOnUiThread {
+                    progressbar_login.visibility = View.GONE
+                }
+
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt]: extra empty line

File: app/src/main/java/com/example/android_demo_kotlin/adapter/RecyclerviewApiAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                withContext(Dispatchers.Main) {
+                    imageOfUser.setImageBitmap(image)
+                    firstNameOfUser.text = usersList[position].firstName
+                    lastNameOfUser.text = usersList[position].lastName
+                    emailOfUser.text = usersList[position].email
+                    layout.setOnClickListener {
+                        usersList[position].id?.let { it1 -> itemClickCallBack(it1) }
+                    }
+                }
+
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/android_demo_kotlin/adapter/RecyclerviewApiAdapter.kt]: code formatting

**Review by Purvesh-Dodiya — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+import java.io.OutputStreamWriter
+import java.net.HttpURLConnection
+import java.net.URL
+import kotlinx.android.synthetic.main.activity_login.btn_login
+import kotlinx.android.synthetic.main.activity_login.editxt_email_login
+import kotlinx.android.synthetic.main.activity_login.editxt_password_login
+import kotlinx.android.synthetic.main.activity_login.progressbar_login
+import org.json.JSONObject
+
+
```
Comment (Purvesh-Dodiya) [inline app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt]: extra empty line

File: app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt (hunk 2 above)
Comment (Purvesh-Dodiya) [reply app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt]: pending

**Review by Purvesh-Dodiya — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/android_demo_kotlin/WebServices/RecyclerWebActivity.kt [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_recycler_web)
+        recyclerview_progressbar.visibility = View.VISIBLE
+        getMethod()
+    }
+
+    private fun getMethod() {
+        GlobalScope.launch(Dispatchers.IO) {
+            val url = URL(BASE_URL + "api/users?page=2")
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/WebServices/RecyclerWebActivity.kt]: Check in whole file for static keys and move to constant

File: app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        with(url.openConnection() as HttpURLConnection) {
+            requestMethod = getString(R.string.request_method)
+            setRequestProperty(getString(R.string.json_content_type), getString(R.string.content_type_json))
+            val writer = OutputStreamWriter(outputStream)
+            writer.write(cred.toString())
+            writer.flush()
+            println("URL : $url")
+            println("cred: $cred")
+            println("Response Code: $responseCode")
+            if (responseCode == 200) {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt]: static number, check in whole file

File: app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+                runOnUiThread {
+                    progressbar_login.visibility = View.GONE
+                }
+                return true
+            } else if (responseCode == 400) {
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/WebServices/LoginActivity.kt]: same

**Review by purvesh-d — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/android_demo_kotlin/adapter/RecyclerviewApiAdapter.kt [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
+        holder.apply {
+            GlobalScope.launch(Dispatchers.IO) {
+                val url = usersList[position].avatar?.let {
+                    URL(it)
+                }
+                val image = BitmapFactory.decodeStream(url?.openConnection()?.getInputStream())
+                withContext(Dispatchers.Main) {
+                    imageOfUser.setImageBitmap(image)
+                    firstNameOfUser.text = usersList[position].firstName
```
Comment (purvesh-d) [inline app/src/main/java/com/example/android_demo_kotlin/adapter/RecyclerviewApiAdapter.kt]: use apply for `usersList[position]`

**Review by purvesh-d — CHANGES_REQUESTED** [review]

**Review by purvesh-d — APPROVED** [review]

**Pull request merged**